src/memex/storage.py

```diff
@@ -108,14 +109,16 @@ def create_annotation(request, data):
             )
 
     # The user must have permission to create an annotation in the group
-    # they've asked to create one in.
-    if data['groupid'] != '__world__':
-        group_principal = 'group:{}'.format(data['groupid'])
-        if group_principal not in request.effective_principals:
+    # they've asked to create one in. If the application didn't configure
+    # a groupfinder we will allow writing this annotation without any
+    # further checks.
+    groupfinder = request.registry.get(GROUPFINDER_KEY)
+    if groupfinder is not None:
```

Comment:
Again, this is ok, and I'd be happy to merge it, but the beauty of using a context object is that it doesn't matter what the object is, so you can set a default groupfinder which just returns an instance of `PermissiveGroupContext`:

```python
class PermissiveGroupContext(object):
    __acl__ = [(security.Allow, security.Everyone, security.ALL_PERMISSIONS)]
```

or something.